packages/integrations/cloudflare/src/index.ts

```diff
@@ -105,17 +195,49 @@ export default function createIntegration(args?: Options): AstroIntegration {
 				_buildConfig = config.build;
 
 				if (config.output === 'static') {
-					throw new Error(`
-  [@astrojs/cloudflare] \`output: "server"\` or \`output: "hybrid"\` is required to use this adapter. Otherwise, this adapter is not necessary to deploy a static site to Cloudflare.
-
-`);
+					throw new Error(STATIC_ERROR);
 				}
 
 				if (config.base === SERVER_BUILD_FOLDER) {
-					throw new Error(`
-  [@astrojs/cloudflare] \`base: "${SERVER_BUILD_FOLDER}"\` is not allowed. Please change your \`base\` config to something else.`);
+					throw new Error(FOLDER_ERROR);
 				}
 			},
+			'astro:server:setup': ({ server }) => {
+				server.middlewares.use(async function middleware(req, res, next) {
```

Comment:
I think it's better to add a new middleware only if `runtimeMode` isn't `'off'`:

```js
if( runtimeMode !== 'off') {
	server.middlewares.use();
}
```